src/datadog.ts

```diff
@@ -63,65 +63,63 @@ export class Datadog extends Construct {
 
     const extractedLambdaFunctions = extractSingletonFunctions(lambdaFunctions);
 
-    if (this.props !== undefined && extractedLambdaFunctions.length > 0) {
-      if (this.props.apiKeySecret !== undefined) {
-        grantReadLambdas(this.props.apiKeySecret, extractedLambdaFunctions);
-      } else if (
-        this.props.apiKeySecretArn !== undefined &&
-        construct !== undefined &&
-        baseProps.grantSecretReadAccess
-      ) {
-        log.debug("Granting read access to the provided Secret ARN for all your lambda functions.");
-        grantReadLambdasFromSecretArn(construct, this.props.apiKeySecretArn, extractedLambdaFunctions);
-      }
+    if (extractedLambdaFunctions.length === 0) {
```

Comment:
1. `this.props` can't be `undefined`, so removing this check.
2. flip the if condition and move the large chunk of code outside the if block